roles/keystone/templates/etc/shibboleth/shibboleth2.xml

```diff
@@ -28,10 +28,10 @@
     <ReplayCache StorageService="mc"/>
     <ArtifactMap StorageService="mc" artifactTTL="180"/>
 
-    <ApplicationDefaults entityID="{{ keystone.federation.sp.k2k.sp_id }}">
-        <Sessions lifetime="28800" timeout="3600" relayState="ss:mem"
+    <ApplicationDefaults entityID="{{ keystone.federation.sp.saml.sp_id }}">
+        <Sessions lifetime="28800" timeout="3600" relayState="ss:mc"
```

Comment:
Yes, the storage ID should have been 'mc' in the first place. I think the reason why this was not caught earlier was because K2K did not require relay state since it only required the ECP protocol and not full Web SSO (which requires relay state during sessions).